src/Datadog.Trace.ClrProfiler.Managed/Integrations/WebRequestIntegration.cs

```diff
@@ -35,11 +39,32 @@ public static class WebRequestIntegration
             TargetMaximumVersion = Major4)]
         public static object GetResponse(object webRequest, int opCode, int mdToken, long moduleVersionPtr)
         {
+            const string methodName = nameof(GetResponse);
+            var webRequestType = webRequest.GetType();
+            Func<object, WebResponse> callGetResponse;
+
+            try
+            {
+                var instrumentedType = webRequest.GetInstrumentedType("System.Net.WebRequest");
+                callGetResponse =
+                    MethodBuilder<Func<object, WebResponse>>
+                        .Start(moduleVersionPtr, mdToken, opCode, methodName)
+                        .WithConcreteType(instrumentedType)
+                        .WithNamespaceAndNameFilters("System.Net.WebResponse")
+                        .Build();
+            }
+            catch (Exception ex)
+            {
+                // profiled app will not continue working as expected without this method
+                Log.ErrorException($"Error retrieving {webRequestType.Name}.{methodName}()", ex);
```

Comment:
We're looking for the method on `System.Net.WebRequest`. `webRequestType` will be a derived type.
```suggestion
                Log.ErrorException($"Error retrieving {instrumentedType .Name}.{methodName}()", ex);
```

Reply:
Would it be valuable to know the instance type for debugging purposes? Should we expand the log to include both?

Reply:
Should we also start logging the ModuleVersionId and mdToken here?

Reply:
For posterity, Colin I'm going to defer the logging of ModuleVersionId and mdToken since it sounds like you're going to tackle that in a follow-up item.